src/mame/tandberg/tdv2100_disp_logic.cpp

```diff
@@ -0,0 +1,962 @@
+	m_beep->add_route(ALL_OUTPUTS, "mono", 1.0);
+
+	RS232_PORT(mconfig, m_rs232, default_rs232_devices, nullptr);
+	m_rs232->rxd_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_rxd_w));
+	m_rs232->dcd_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_dcd_w));
+	m_rs232->dsr_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_dsr_w));
+	m_rs232->cts_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_cts_w));
+	m_rs232->ri_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_ri_w));
+	AY51013(mconfig, m_uart);
+	m_uart->write_so_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_txd_w));
+	m_uart->write_dav_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::uart_rx));
+	m_uart->write_dav_callback().append(FUNC(tandberg_tdv2100_disp_logic_device::check_rs232_rx_error));
+	m_uart->write_tbmt_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::uart_tx));
+	m_uart->write_pe_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::check_rs232_rx_error));
+
+	CLOCK(mconfig, m_uart_clock, 1);
+	// NOTE: Frequency set with the rest of the UART settings
+	m_uart_clock->signal_handler().set(m_uart, FUNC(ay51013_device::write_rcp));
+	m_uart_clock->signal_handler().append(m_uart, FUNC(ay51013_device::write_tcp));
+}
+
+static INPUT_PORTS_START( tdv2115l )
+
+	PORT_START("sw_invert_video")
+		PORT_CONFNAME(0x1, 0x1, "INVERSE VIDEO")
+			PORT_CONFSETTING(0x0, "Toggled in")
+			PORT_CONFSETTING(0x1, "Toggled out")
+
+	PORT_START("sw_rs232_baud")
+		PORT_CONFNAME(0x7, 0x6, "SPEED SELECT (Baud-rate)")                                                             PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+			PORT_CONFSETTING(0x0, "0: 110")
+			PORT_CONFSETTING(0x1, "1: 300")
+			PORT_CONFSETTING(0x2, "2: 600")
+			PORT_CONFSETTING(0x3, "3: 1200")
+			PORT_CONFSETTING(0x4, "4: 2400")
+			PORT_CONFSETTING(0x5, "5: 4800")
+			PORT_CONFSETTING(0x6, "6: 9600")
+			PORT_CONFSETTING(0x7, "7: 19200")
+
+	PORT_START("sw_rs232_settings")
+		PORT_CONFNAME(0x01, 0x01, "NO PARITY")              PORT_CONFSETTING(0x01, "YES") PORT_CONFSETTING(0x00, "NO")  PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+		PORT_CONFNAME(0x02, 0x02, "EVEN PARITY")            PORT_CONFSETTING(0x02, "YES") PORT_CONFSETTING(0x00, "NO")  PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+		PORT_CONFNAME(0x04, 0x04, "TWO STOP BITS")          PORT_CONFSETTING(0x04, "YES") PORT_CONFSETTING(0x00, "NO")  PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+		PORT_CONFNAME(0x08, 0x08, "EXT. ECHO")              PORT_CONFSETTING(0x08, "YES") PORT_CONFSETTING(0x00, "NO")
+		PORT_CONFNAME(0x10, 0x00, "AUTO RFS (RTS)")         PORT_CONFSETTING(0x10, "YES") PORT_CONFSETTING(0x00, "NO")
+		PORT_CONFNAME(0x20, 0x00, "AUTO DSR")               PORT_CONFSETTING(0x20, "YES") PORT_CONFSETTING(0x00, "NO")
```

Comment:
Please put one setting per line.  Also, can you use the default yes/no strings, and order the settings so no comes before yes, to give some kind of consistency?

How do the parity settings interact?  Turning on no parity at the same time as even parity doesn’t make a great deal of sense.  Should it be a 2-bit field so the four possible states can be labelled clearly?

What happens if you turn two stop bits off?  Would it be better to just call it stop bits, and then have the number of stop bits for the two setting values?

Reply:
I tried to be consistent with the printed labels on the back casing of the machine here, but I can change to default and more descriptive strings if that is preferred. Most of these settings were originally only described in detail through the user-manual, for instance see page 3: 

https://www.technikum29.de/de/geraete/Siemens_6610/Siemens_6610_PDFs/Siemens_6610_Wartungshandbuch_Teil_2_Einschaltung_Bruecken_und_Einstellungen.pdf

(The 6.610 is an OEM version of the TDV-2114, which uses the same terminal module as the TDV-2115. They both have most of the same switches exposed.)